### DISCUSSION TITLE
PinJoints are liable to separating under stress

### DISCUSSION THREAD

**Discussion by AyOhEe** [godot-jolt/godot-jolt#921] [Questions] [answered]
I'm attempting to create a physical model of an arm, and I'm using PinJoints along with `PhysicsServer3D.pin_joint_set_local_a [ or b]` to control where the bodies should be relative to each other. This works great, however, when the hand is gripping a frozen body and the shoulder is moving away, the elbow and wrist joints separate due to the intense force.

Under normal circumstances I'd increase the bias of the joints, however under jolt this isn't possible and there is no substitute that I can see. Is there anything I can do to increase the strength of PinJoints?

**Comment by mihe** [comment, marked as answer]
If you use `JoltPinJoint3D` instead of `PinJoint3D` you'll have two properties called `solver_velocity_iterations` and `solver_position_iterations` that can be increased (at the cost of performance) to lessen the effect of this, but you'll never be able to mitigate it entirely.

If you absolutely must use `PinJoint3D` you can also change this using `JoltPhysicsServer3D.joint_set_solver_velocity_iterations` and `JoltPhysicsServer3D.joint_set_solver_position_iterations` respectively, as both joints share the same "backend".

I believe Jolt's stabilization also plays some role in the effect you're seeing, which can be lessened by increasing the "Position Correction" project setting, but then you're sacrificing the stability of your entire simulation, which may not be advisable.

**Reply by AyOhEe** [reply]
Upping `solver_velocity_iterations` and `solver_position_iterations` helped a lot, thanks!
